src/visualizers/VisualizerHistogram.ts

```diff
@@ -0,0 +1,211 @@
+            }
+        }
+        return largest_value
+    }
+
+    // Create an array with the number of factors of
+    // each element at the corresponding index of the array.
+    factorArray(): number[] {
+        const factorArray = []
+        for (
+            let i = Math.max(this.firstIndex, this.seq.first);
+            i < Math.min(this.terms, this.seq.last);
+            i++
+        ) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [, power] of factors) {
+                    counter += Number(power)
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding index.
+    orderedFactorArray(): number[] {
+        const orderedFactorArray = []
+        const factorArray = this.factorArray()
+        for (
+            let i = 0;
+            i < natlog(this.largestValue()) / natlog(2) + 1;
+            i++
+        ) {
+            orderedFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            orderedFactorArray[factorArray[i]]++
+        }
+
+        return orderedFactorArray
+    }
+
+    // Create an array to change the bin size of the
+    // histogram to what the user asked for.
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const orderedFactorArray = this.orderedFactorArray()
+        for (
+            let i = 0;
+            i < Math.ceil(orderedFactorArray.length / this.binSize);
+            i++
+        ) {
+            binFactorArray.push(0)
+        }
+
+        let index = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            for (let j = 0; j < this.binSize; j++) {
+                if (index + j < orderedFactorArray.length) {
+                    binFactorArray[i] += orderedFactorArray[index + j]
+                }
+            }
+            index += this.binSize
+        }
+
+        return binFactorArray
+    }
+
+    // Create a number that represents how
+    // many pixels wide each bin should be.
+    binWidth(): number {
+        const binFactorArray = this.binFactorArray()
+        let binWidth = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] != 0) {
+                binWidth = i
+            }
+        }
+        return 750 / (binWidth + 1)
+    }
+
+    // Create a number that represents how many pixels high
+    // each increase of one in the bin array should be.
+    height(): number {
+        const binFactorArray = this.binFactorArray()
+        let height = binFactorArray[0]
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] > height) {
+                height = binFactorArray[i]
+            }
+        }
+        return 750 / height
```

Comment:
Same issue about the "magic number" 750...

It's also a bit misleading to name your variable "height" since then it seems like that would be the value you want to return from a function called "height". Maybe rename it to "tallest" or something like that?

Reply:
This function returns a number that I multiply by the number of elements in each bin to convert it to it's height in pixels.

Reply:
Sorry if I was unclear. The function name is just fine. It's the variable name that is confusing. Naming that variable "height" suggests that it will be the return value of the function, which it is not. So please name the variable something that indicates what that variable actually means. (It is the maximum number of items in a bin, which is why I suggested "tallest". But if another name makes more sense to you, feel free to use something else. Please just don't name it after the function when it is not the value being computed by the function. Thanks.)

Reply:
Okay, That makes sense. Thank you.